test/e2e/framework/test_context.go

```diff
@@ -164,6 +164,9 @@ type TestContextType struct {
 
 	// The configuration of NodeKiller.
 	NodeKiller NodeKillerConfig
+
+	// The IP Family of the cluster
+	IPFamily string
```

Comment:
/hold 

IMO we need to have auto-detection for this type of behavior. 

/assign @timothysc.

Reply:
@timothysc I've added the variable here so is available to all the text, but it's populated using a detection function at the beginning of the testing, in `test/e2e/e2e.go`

Reply:
there is so much random stuff in the context and that's a prime example of organic growth.

seems to me we should not add more fields to the context or flags to the CLI, or at least that's what i understood was the plan. it might slightly increase the test suite runtime if the detection func  is called multiple times, but it still feels better.

Reply:
I didn't know how to make this variable available to all the tests, would be happy to change it if you give me some advice.
 I don't think that running the function multiple times is a good option, at least without caching, since the IP family of the cluster is immutable.

Reply:
> I didn't know how to make this variable available to all the tests, would be happy to change it if you give me some advice.

the context is the dumping ground for all shared fields, which wrong. but AFAIK there is no other way to share properties right now.

> I don't think that running the function multiple times is a good option, at least without caching, since the IP family of the cluster is immutable.

in my view the API call should be pretty cheap, but yes if the `kubernetes` service is not changing it's IP family then it is a bit of a waste. i'd still go calling it multiple times and not depending on the context though.

/assign @oomichi 
^ for opinion here too.

Reply:
Sorry for my late response.
The cost of the API call `c.CoreV1().Services(metav1.NamespaceDefault).Get("kubernetes", metav1.GetOptions{})` can be cheap then it doesn't matter if we don't use cache here.
I saw @neolit123's point because I am not sure the string `ipv4` / `ipv6` could be nice for the global context for the code readability.
How about adding comment at least like
```
	// The IP Family of the cluster ("ipv4" or "ipv6")
	IPFamily string
```
even if not changing the code?

Reply:
done